service/src/main/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityUserService.java

```diff
@@ -1,71 +1,75 @@
+    public LocalAuthorityUserService(OrganisationService organisationService,
                                      AuthTokenGenerator authTokenGenerator,
                                      CaseUserApi caseUserApi,
                                      IdamClient idamClient,
                                      SystemUpdateUserConfiguration userConfig) {
-        this.caseAccessApi = caseAccessApi;
-        this.localAuthorityUserLookupConfiguration = localAuthorityUserLookupConfiguration;
+        this.organisationService = organisationService;
         this.authTokenGenerator = authTokenGenerator;
         this.caseUserApi = caseUserApi;
         this.client = idamClient;
         this.userConfig = userConfig;
     }
 
-    public void grantUserAccessWithCaseRole(String caseId, String caseLocalAuthority) {
-        findUserIds(caseLocalAuthority).stream()
-            .forEach(userId -> {
+    public void grantUserAccessWithCaseRole(String authorisation,
+                                            String userId,
+                                            String caseId,
+                                            String caseLocalAuthority) {
+        List<String> userIds = findUserIds(authorisation, caseLocalAuthority);
+
+        Stream.concat(userIds.stream(), Stream.of(userId))
+            .distinct()
+            .forEach(id -> {
                 try {
                     String authentication = client.authenticateUser(userConfig.getUserName(), userConfig.getPassword());
-                    caseUserApi.updateCaseRolesForUser(authentication, authTokenGenerator.generate(), caseId, userId,
-                        new CaseUser(userId, caseRoles));
+                    caseUserApi.updateCaseRolesForUser(authentication, authTokenGenerator.generate(), caseId, id,
+                        new CaseUser(id, caseRoles));
 
-                    logger.info("Added case roles {} to user {}", caseRoles, userId);
+                    logger.info("Added case roles {} to user {}", caseRoles, id);
                 } catch (Exception exception) {
                     logger.warn("Error adding case roles {} to user {}",
-                        caseRoles, userId, exception);
+                        caseRoles, id, exception);
                 }
             });
     }
 
-    private List<String> findUserIds(String localAuthorityCode) {
-        List<String> userIds = localAuthorityUserLookupConfiguration.getUserIds(localAuthorityCode);
-
-        if (userIds.isEmpty()) {
-            throw new NoAssociatedUsersException("No users found for the local authority '" + localAuthorityCode + "'");
+    private List<String> findUserIds(String authorisation, String localAuthorityCode) {
+        try {
+            return organisationService
+                .findUserIdsInSameOrganisation(authorisation, localAuthorityCode);
+        } catch (Exception e) {
+            log.warn("Exception while looking for users within the same LA. "
```

Comment:
I'd say not quite right place to say who gets access but no time to make it better.